api/resources_portal/views/organization_invitations.py

```diff
@@ -0,0 +1,103 @@
+
+
+class OrganizationInvitationSerializer(serializers.ModelSerializer):
+    requester = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    request_reciever = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    organization = serializers.PrimaryKeyRelatedField(queryset=Organization.objects.all())
+
+    class Meta:
+        model = OrganizationInvitation
+        fields = (
+            "id",
+            "created_at",
+            "updated_at",
+            "status",
+            "invite_or_request",
+            "organization",
+            "request_reciever",
+            "requester",
+        )
+
+
+class OrganizationInvitationViewSet(viewsets.ModelViewSet):
+    queryset = OrganizationInvitation.objects.all()
+    serializer_class = OrganizationInvitationSerializer
+
+    def update_organizations(self, new_status, invitation):
+        old_status = invitation.status
+        if old_status == "PENDING" and new_status == "ACCEPTED":
+            invitation.organization.members.add(invitation.requester)
+            # TODO: Add notification of acceptance
+        elif old_status == "PENDING" and new_status == "REJECTED":
+            pass
+            # TODO: Add notification of rejection
+        elif old_status == "PENDING" and new_status == "INVALID":
+            pass
+            # TODO: Add notification of invalid request reciever
+
+    def create(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if not OrganizationInvitationSerializer(data=request.data).is_valid():
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        request_reciever = User.objects.get(pk=request.data["request_reciever"])
+        organization = Organization.objects.get(pk=request.data["organization"])
+
+        if request.data["invite_or_request"] == "INVITE" and not request_reciever.has_perm(
+            "add_members_and_manage_permissions", organization
+        ):
+            return Response(
+                f"{request_reciever} does not have permission to add members",
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        return super(OrganizationInvitationViewSet, self).create(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+        invitation = OrganizationInvitation.objects.get(pk=kwargs["pk"])
+
+        requester_accepting = (
+            request.user == invitation.requester and invitation.invite_or_request == "INVITE"
+        )
+        request_reciever_approving = (
+            request.user == invitation.request_reciever
+            and invitation.invite_or_request == "REQUEST"
+        )
+
+        # Xor operator
+        if not requester_accepting ^ request_reciever_approving:
```

Comment:
```suggestion
        if not (requester_accepting or request_reciever_approving):
```
I feel like XOR is generally less clear unless it's important that one and only one condition is true. Both of the conditions you're testing cannot be true at the same time. We're really just testing that at least one of them is true.